### ISSUE TITLE
[BUG/ISSUE] Bug in henry_mod/CALC_HEFF

### ISSUE THREAD

**Issue by viral211** [geoschem/geos-chem#1001]
### Problem
The calc_heff routine calculates the effective Henry's law constant accounting for acid dissociation. calc_heff is called from wetscav_mod/compute_l2g and gckpp_hetrates/compute_l2g_local. 
In the current code (version 13.2) we compute Heff only when the pKa is greater than 0, but pKa can be negative. 
https://github.com/geoschem/geos-chem/blob/ba58da4a6e447e47714fe5e7da488d15432a6fdc/GeosUtil/henry_mod.F90#L196-L200
This condition was set to bypass the calculation when pKa is not defined, which is true for all of the species in the species database, since the Henry's law constants specified in the species database for use in wetscav_mod are already the effective ones. So calc_heff does not need to be called from wetscav_mod/compute_l2g at all. However, when calc_heff is called from gckpp_hetrates/compute_l2g_local, pKa is defined and is negative. This returns an incorrect Heff, which affects the partitioning of acids (HCl and HBr) to cloud droplets.

### Proposed solution:
(i) Remove the pKa > 0 condition in calc_heff, and (ii) remove the call to calc_heff in wetscav_mod/compute_l2g.

**Comment by yantosca** [comment]
Thanks @viral211 .  There was some discussion a while back about not adjusting the Henry's law constant KH for pH but I can't remember why.  In either case we should use consistent code in both places.

**Comment by viral211** [comment]
Thanks @yantosca. To use consistent code in both places, we could set the condition pKa>-100, since we use -999 to represent the missing value.

**Comment by yantosca** [comment]
Thanks @viral.  I think we can get this into 13.4.0 pretty easily.

**Comment by yantosca** [comment]
The other thing we can do is to try to pull out the code so that it is only kept in 1 place.  I think we replicated it in the old gckpp_Hetrates to avoid bringing in wetdep code into chemistry code.  We can maybe make a separate folder and compile the henry_mod first so that it can be used in wetscav and in hetchem.

**Comment by viral211** [comment]
That would be better. The henry_mod routines could also be used in sulfate_mod, where we do this calculation numerous times when calculating cloud water pH. It would make the code cleaner.

**Comment by viral211** [comment]
Hi @yantosca. I believe `V_tot` corresponds to `H2OLIQ`, which is the cloud liquid water content. Whereas `L2G` is `H_eff * H2OLIQ`. Note that `H_eff` is dimensionless in this equation, not in the more commonly used units of M/atm.

**Comment by yantosca** [comment]
Thanks @viral211.  Will fix that.

**Comment by yantosca** [comment]
I think this is now correct.  Thanks @viral211!
```F90
    ! Note from Viral Shah (06 Dec 2021):
    !   I believe V_tot corresponds to H2OLIQ, which is the cloud liquid
    !   water content. Whereas L2G is H_eff * H2OLIQ. Note that H_eff is
    !   dimensionless in this equation, not in the more commonly used units
    !   of M/atm.

    ! Chloride (mol/L)
    CALL Compute_L2G_Local( K0     =  1.0_dp,     CR = 9000.0_dp,            &
                            pKa    = -6.3_dp,     TK = TEMP,                 &
                            H2OLIQ =  V_tot,      pH = H%pHCloud,            &
                            L2G    =  L2G                                   )
    F_L = L2G / ( 1.0_dp + L2G )
    Cl_conc = F_L * HCl / (V_tot * H%AVO * 1.0e-3_dp)

    ! Bromide (mol/L)
    CALL Compute_L2G_Local( K0     =  7.5e-1_dp,  CR = 10200.0_dp,           &
                            pKa    = -9.0_dp,     TK = TEMP,                 &
                            H2OLIQ =  V_tot,      pH = H%pHCloud,            &
                            L2G    =  L2G                                   )
    F_L = L2G / ( 1.0_dp + L2G )
    Br_conc = F_L * HBr / ( V_tot * H%AVO * 1.0e-3_dp )
```

**Comment by yantosca** [comment]
Integration tests for GCC passed.  Waiting on GCHP results.

**Comment by yantosca** [comment]
PR #1042 has now been merged into dev and tagged with 13.4.0-alpha.5.  We can now close out this issue.
